go-ssb-bindings/vendor/go.cryptoscope.co/ssb/internal/muxmux/async.go

```diff
@@ -0,0 +1,42 @@
+package muxmux
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"go.cryptoscope.co/muxrpc"
+)
+
+type AsyncFunc func(context.Context, *muxrpc.Request) (interface{}, error)
+
+func (af AsyncFunc) HandleAsync(ctx context.Context, r *muxrpc.Request) (interface{}, error) {
+	return af(ctx, r)
+}
+
+type AsyncHandler interface {
+	HandleAsync(context.Context, *muxrpc.Request) (interface{}, error)
+}
+
+type asyncStub struct {
+	logger log.Logger
+
+	h AsyncHandler
+}
+
+func (hm asyncStub) HandleCall(ctx context.Context, req *muxrpc.Request, edp muxrpc.Endpoint) {
```

Comment:
It took me a moment to understand why this file exists, it could use a line or two of description <3

Reply:
yes, this was a muxer v2 experiment, tried on ssb-friends.

The first interface for registering handler functions is just super repetitive and clumsy to use.